rflx/pyrflx/typevalue.py

```diff
@@ -1351,6 +1352,8 @@ def __simplified(self, expr: Expr) -> Expr:
         def subst(expression: Expr) -> Expr:
             if expression in {TRUE, FALSE}:
                 return expression
+            if isinstance(expression, Not):
+                return -expression
```

Comment:
What is the use case for this code?

Reply:
Which test cases fail when this code is removed?

Reply:
`test_no_verification_icmp_checksum_with_not_operator` in `tests/integration/pyrflx_test.py`